pkg/kubelet/kubelet_pods_test.go

```diff
@@ -149,10 +152,54 @@ fe00::2	ip6-allrouters
 203.0.113.1	podFoo.domainFoo	podFoo
 `,
 		},
+		{
+			"203.0.113.1",
+			"podFoo",
+			"domainFoo",
+			[]v1.HostAlias{
+				{IP: "123.45.67.89", Hostnames: []string{"foo", "bar", "baz"}},
+			},
+			`# Kubernetes-managed hosts file.
+127.0.0.1	localhost
+::1	localhost ip6-localhost ip6-loopback
+fe00::0	ip6-localnet
+fe00::0	ip6-mcastprefix
+fe00::1	ip6-allnodes
+fe00::2	ip6-allrouters
+203.0.113.1	podFoo.domainFoo	podFoo
+123.45.67.89	foo
+123.45.67.89	bar
+123.45.67.89	baz
+`,
+		},
+		{
+			"203.0.113.1",
+			"podFoo",
+			"domainFoo",
+			[]v1.HostAlias{
+				{IP: "123.45.67.89", Hostnames: []string{"foo", "bar", "baz"}},
+				{IP: "456.78.90.123", Hostnames: []string{"park", "doo", "boo"}},
+			},
+			`# Kubernetes-managed hosts file.
+127.0.0.1	localhost
+::1	localhost ip6-localhost ip6-loopback
+fe00::0	ip6-localnet
+fe00::0	ip6-mcastprefix
+fe00::1	ip6-allnodes
+fe00::2	ip6-allrouters
+203.0.113.1	podFoo.domainFoo	podFoo
+123.45.67.89	foo
+123.45.67.89	bar
+123.45.67.89	baz
+456.78.90.123	park
+456.78.90.123	doo
+456.78.90.123	boo
```

Comment:
should we add validation that there are not duplicate hostnames in case someone adds:

```golang
[]v1.HostAlias{
	{IP: "123.45.67.89", Hostnames: []string{"foo", "bar"}},
	{IP: "456.78.90.123", Hostnames: []string{"foo", "bar"}},
}
```

which results in

```
123.45.67.89	foo
123.45.67.89	bar
456.78.90.123	foo
456.78.90.123	bar
```